lib/Dialect/TTIR/Analysis/ShardSolver.cpp

```diff
@@ -14,7 +21,8 @@ ShardSolver::ShardSolver(
     const llvm::DenseMap<Operation *, std::vector<LayoutAttr>> &legalLayouts,
     const std::vector<ShardSpec> &shardSpecs,
     const llvm::DenseSet<Operation *> &shardedOps,
-    const unsigned usableL1CacheSize)
+    const unsigned usableL1CacheSize,
+    llvm::StringMap<InputLayoutOverrideParams> *inputLayoutOverrides)
```

Comment:
Lets not do this - pass override data directly to inner structures of Optimizer analysis. Convert this in early optmizer phase to vector of edges and just pass in list of edges that need to be resharded.